gui/default/syncthing/core/syncthingController.js

```diff
@@ -1210,11 +1226,40 @@ angular.module('syncthing.core')
             return result;
         };
 
+        $scope.settingsModified = function () {
+            // angular.equals does not check nested objects/arrays, so do this one differently.
```

Comment:
angular.equals is documented to do deep (recursive) comparison - are you sure it's not something else that's tripping up your usage of it?